Link toolbar movability to floating status

ferdnyc opened the pull request:
This PR, targeting @JacksonRG 's PR #4656, will link freezing of the main toolbar to its floating status, so that we don't create an immovable floating toolbar when we freeze the window.

The toolbar's `topLevelChanged` signal (which indicates floating status) is connected to a slot that applies current frozen status if it's docked, and ensures it's always movable if it's floating. The same slot ensures that if a floating toolbar gets docked while the window is frozen, it will freeze in place and no longer be movable until the freeze is deactivated.

I also slightly re-worked how initial freeze state is applied when loading settings, as well as how the docks get frozen/unfrozen, to reduce code duplication.

Commit:
Freeze and unfreeze toolbar along with other windows.

Commit:
Fix font size issues in title editor

Commit:
Split if-else one liner into multiple lines for clarity

Commit:
Default font_size_ratio to 1 if existing font size is invalid

jonoomph committed:
Merge pull request #4660 from hexcone/develop

Fix font size issues in title editor

MartinThoma committed:
STY: Using bool on comparisons is not necessary (#4653)

ferdnyc committed:
Link toolbar movability to floating status

JacksonRG commented on src/windows/main_window.py:
```diff
@@ -2060,39 +2061,41 @@ def showDocks(self, docks):
+                QDockWidget.DockWidgetFloatable
+                | QDockWidget.DockWidgetMovable)
+            if dock is not self.dockTimeline:
+                features |= QDockWidget.DockWidgetClosable
```
I had questions whether a `|=` was really called for. But with how Qt stores features, it looks like best approach.

ferdnyc replied:
Yeah, all the alternatives are uglier / more redundant.Either, more like the old code,
```python3
if dock is self.dockTimeline:
    features = (
        QDockWidget.DockWidgetFloatable
        | QDockWidget.DockWidgetMovable)
else:
    features = (
        QDockWidget.DockWidgetFloatable
        | QDockWidget.DockWidgetMovable
        | QDockWidget.DockWidgetClosable)
dock.setFeatures(features)
```
(or just two separate calls to `dock.setFeatures()` directly, without a `features` variable...)

Or else, something weird like:
```python3
features = (
    QDockWidget.DockWidgetFloatable
    | QDockWidget.DockWidgetMovable)
if dock is self.dockTimeline:
    dock.setFeatures(features)
else:
    dock.setFeatures(features | QDockWidget.DockWidgetClosable)
```

JacksonRG commented on src/windows/main_window.py:
```diff
@@ -3061,6 +3065,10 @@ def __init__(self, *args):
         self.SelectionAdded.connect(self.addSelection)
         self.SelectionRemoved.connect(self.removeSelection)
 
+        # Connect toolbar float changes to movability/frozenness
```
Imagining reading this without reading `freezeMainToolBar`, the function name implies that we freeze the tool bar whenever it gets docked/undocked. Including "Freeze toolbar IF the view is frozen" in the comment might clarify.

ferdnyc replied:
Fair enough. I sort of take it as an _invitation_ to read `freezeMainToolBar` to learn the details (I mean, there's a lot that's implicit here, including how the `topLevelChanged` signal works), but point taken. How about...
```suggestion
        # Ensure toolbar is movable when floated (even with docks frozen)
```
That's not the _entirety_ of what `freezeMainToolBar()`  does (since it also freezes it when _re_-docked), but it's the main reason it exists.

JacksonRG reviewed (approved):
Just a comment suggestion. The code looks awesome!

ferdnyc committed:
Presets: Add MKV (h.264) sw/hw presets (#4637)

- Clone every `format_mp4_x64*.xml` preset into an equivalent
  `format_mkv_x264*.xml` preset, with only the title and
  container format changed.

ferdnyc committed:
Merge branch 'include-toolbar-in-freeze' into toolbar-freezing

ferdnyc commented:
Huh. Looks fine, actually.

![image](https://user-images.githubusercontent.com/538020/152856032-2147e713-0ad9-44aa-a9a3-6e82633efebf.png)

Well, cool.
